src/components/PeopleManagement/data/hooks/useEnterpriseMembersTableData.js

```diff
@@ -0,0 +1,58 @@
+import {
+  useCallback, useMemo, useState,
+} from 'react';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import { logError } from '@edx/frontend-platform/logging';
+import debounce from 'lodash.debounce';
+
+import LmsApiService from '../../../../data/services/LmsApiService';
+
+const useEnterpriseMembersTableData = ({ enterpriseId }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [enterpriseMembersTableData, setEnterpriseMembersTableData] = useState({
+    itemCount: 0,
+    pageCount: 0,
+    results: [],
+  });
+  const fetchEnterpriseMembersData = useCallback((args) => {
+    const fetch = async () => {
+      try {
+        setIsLoading(true);
+        const options = {};
+        args.filters.forEach((filter) => {
+          const { id, value } = filter;
+          if (id === 'name') {
+            options.user_query = value;
+          }
+        });
+
+        options.page = args.pageIndex + 1;
+        const response = await LmsApiService.fetchEnterpriseCustomerMembers(enterpriseId, options);
+        const data = camelCaseObject(response.data);
+        setEnterpriseMembersTableData({
+          itemCount: data.count,
+          pageCount: data.numPages ?? Math.floor(data.count / options.pageSize),
+          results: data.results,
+        });
+      } catch (error) {
+        logError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    fetch();
```

Comment:
Because we are displaying cards in the table, every keystroke causes the table to rerender and shifts the layout which can disrupt the user experience.
https://github.com/user-attachments/assets/470b5ca5-4118-40dd-87ff-1ad103788717

since we are already debouncing, I would suggest to also add in a minimum query string limit, let's say more than 2 characters so that we're only going to make requests when the user types in more than 2 characters. I think we can do something like this: 

```
    if (args.filters.length && args.filters[0].value.length > 2) {
      fetch()
    } else if(!args.filters.length) {
      fetch()
    }
```